### ISSUE TITLE
react/jsx-handler-names should not flag arrow functions

### ISSUE THREAD

**Issue by astorije** [yannickcr/eslint-plugin-react#2586]
Given the following configuration:

```yml
react/jsx-handler-names:
  - error
  - eventHandlerPrefix: '(handle|on|set)'
    checkLocalVariables: true
```

The following code:

```tsx
<Button onClick={() => setVisible(!visible)}>
  {visible ? 'Hide' : 'Show'}
</Button>
```

fails linting with the following message:

```
Handler function for onClick prop key must begin with '(handle|on|set)'  react/jsx-handler-names
```

We believe `react/jsx-handler-names` should not trigger on arrow functions, because there is no way to address the handler naming violation without refactoring into a variable or similar (which, while probably valid, is another rule/problem).
Thoughts?

**Comment by ljharb** [comment]
At the least, that seems like a reasonable option to add.

**Comment by aaronnunez95** [comment]
I agree with this correction because it would help a lot to work with react hooks.

**Comment by dididy** [comment]
Could I resolve this issue? I'm working on it.

**Comment by ljharb** [comment]
@dididy yes, anything with the "help wanted" label that doesn't already have a linked PR is up for grabs :-)

**Comment by roshcagra** [comment]
Any update on this? Glad to help!

**Comment by dididy** [comment]
@roshcagra Could you review my Pull Request? https://github.com/yannickcr/eslint-plugin-react/pull/2761
